### PR TITLE
Webdriverio Test Download Behavior

### PR DESCRIPTION
**Pull request by M-Hammad-Faisal**
# [Document how to set download behavior across browser #12105](https://github.com/webdriverio/webdriverio/issues/12105)

## Changes Made

- [x] Added Safari download behavior.
- [x] Added Chrome download behavior.
- [x] Added Firefox download behavior.
- [x] Added Microsoft Edge download behavior.

## Testing

- [x] Added a test case to download 1KB file.

## Checklist

- [x] Updated documentation
- [x] Ran tests locally for each browser
- [x] Checked for code conflicts

### CODE REVIEW

File: testDownloadBehavior/example.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    const downloadDirectory = __dirname;
+
+    const byteCount = await $("#byteCount");
+    await byteCount.scrollIntoView();
+    await byteCount.setValue("1");
+
+    const downloadButton = await $("div.form-row button");
+    await downloadButton.click();
+
+    await browser.pause(2000);
```
Comment (christian-bromann) [inline testDownloadBehavior/example.js:29]: Why are we pausing here? Can we use `waitUntil` here instead ?

File: testDownloadBehavior/README.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+To test download behavior in **Apple Safari**, use these capabilties.
+
+```javascript
+{
+    browserName: "safari",
+}
+```
+
+## Test Download Behavior
```
Comment (christian-bromann) [inline testDownloadBehavior/README.md:106]: Let's have an NPM script to test for each browser in each section

File: testDownloadBehavior/README.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+```javascript
+{
+    browserName: "safari",
+    "safari.safariOptions":{
+        dataDir: "/path/to/downloads/directory"
+    }
+}
+```
+
+But it **didn't** work out.
```
Comment (christian-bromann) [inline testDownloadBehavior/README.md:81]: No need to document this then, it is fine if we just say that Safari doesn't support an defining download dirs through WebDriver capabilities

**Review by christian-bromann — COMMENTED** [review]
Let's add this example to our CI/CD pipeline in `.github/workflows/test.yml`

File: wdio.conf.js [hunk 4]
```diff
@@ -1,302 +1,342 @@
+import path from "path";
```
[the remark this answers is not in the document]
Comment (christian-bromann) [reply wdio.conf.js:1]: I'd suggest to create a `wdio.conf.js` within `testDownloadBehavior` that imports the base config and only adds the important changes, e.g. capabilities with download behavior settings, these you can then reference in the docs.

File: testDownloadBehavior/README.md (hunk 2 above)
Comment (christian-bromann) [reply testDownloadBehavior/README.md:106]: So maybe we can have apply caps for browser if a certain env variable is set and run all browser if this environment variable is not provided.

File: wdio.conf.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (M-Hammad-Faisal) [reply wdio.conf.js:1]: resolved.

File: testDownloadBehavior/example.js (hunk 1 above)
Comment (M-Hammad-Faisal) [reply testDownloadBehavior/example.js:29]: ok.

File: testDownloadBehavior/README.md (hunk 2 above)
Comment (M-Hammad-Faisal) [reply testDownloadBehavior/README.md:106]: I have used EXAMPLE_BROWSER env variable.

Comment (M-Hammad-Faisal) [reply testDownloadBehavior/README.md:106]: am i doing it the way you asked, right?, to make a separate npm script for it.

**Review by christian-bromann — APPROVED** [review]
Awesome, thanks 👍

**Pull request merged**